subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/ThirdPartyGradleModuleMetadataSmokeTest.groovy

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.smoketests
+
+import groovy.json.JsonSlurper
+import org.gradle.integtests.fixtures.ToBeFixedForInstantExecution
+import org.gradle.internal.os.OperatingSystem
+import org.gradle.testkit.runner.BuildResult
+
+class ThirdPartyGradleModuleMetadataSmokeTest extends AbstractSmokeTest {
+
+    /**
+     * Everything is done in one test to safe execution time.
+     * Running the producer build takes ~2min.
+     */
+    @ToBeFixedForInstantExecution
+    def 'produces expected metadata and can be consumed'() {
+
+        given:
+        BuildResult result
+        def arch = OperatingSystem.current().macOsX ? 'MacosX64' : 'LinuxX64'
+        useSample("gradle-module-metadata-example")
+        def latestAndroidRelease = TestedVersions.androidGradle.versions[-2]
```

Comment:
This might not pick the "latest release" depending on the available AGP versions.

You can pick the latest of a given AGP "generation", e.g. `3.6.x`:
```groovy
        def latestAndroidRelease = AGP_VERSIONS.getLatestOfMinor("3.6")
```
It'll currently use `3.6.0-rc02`.

See `gradle/dependency-management/agp-versions.properties`.

If you really want the "latest stable release" you'd need to augment `AndroidGradlePluginVersions` with such a query.

Reply:
Thanks for the explanation @eskatos. I think `getLatestOfMinor` works for me.